flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/ShowCreateUtil.java

```diff
@@ -87,6 +89,9 @@ public static String buildShowCreateTableRow(
                     String.format(
                             "SHOW CREATE TABLE is only supported for tables, but %s is a view. Please use SHOW CREATE VIEW instead.",
                             tableIdentifier.asSerializableString()));
+        } else if (table.getTableKind() == CatalogBaseTable.TableKind.MATERIALIZED_TABLE) {
+            return buildShowCreateMaterializedTableRow(
```

Comment:
there appears to be duplicated code in the new materialized table method and the existing table code, could we refactor to remove the duplication.

Reply:
could we have a junit for the new materialised table show code please.

Reply:
Thank you very much for your evaluation. I will refactor this code as soon as possible and provide the corresponding test cases